docs/native-modules-android.md

```diff
@@ -1,115 +1,128 @@
+> The React Native team is currently working on a re-architecture of the NativeModule system. This new system is called TurboModules, and it will help facilitate more efficient type-safe communication between JavaScript and native, without relying on the React Native bridge. It will also enable new extensions that weren't possible with the legacy NativeModule system. You can read more about it here. Throughout these docs we have added notes around parts of Native Modules that will change in the TurboModules release and how you can best prepare for a smooth upgrade to TurboModules.
 
-### Enable Gradle
+### Setup
 
-If you plan to make changes in Java code, we recommend enabling [Gradle Daemon](https://docs.gradle.org/2.9/userguide/gradle_daemon.html) to speed up builds.
+To get started, open up the Android project within your React Native application in Android Studio. You can find your Android project here within a React Native app:
 
-## The Toast Module
+<figure>
+  <img src="/docs/assets/native-modules-android-open-project.png" width="500" alt="Image of opening up an Android project within a React Native app inside of Android Studio." />
+  <figcaption>Image of where you can find your Android project</figcaption>
+</figure>
 
-This guide will use the [Toast](http://developer.android.com/reference/android/widget/Toast.html) example. Let's say we would like to be able to create a toast message from JavaScript.
+We recommend using Android Studio to write your native code. Android studio is an IDE built for Android development and using it will help you resolve minor issues like code syntax errors quickly.
 
-We start by creating a native module. A native module is a Java class that usually extends the `ReactContextBaseJavaModule` class and implements the functionality required by the JavaScript. Our goal here is to be able to write `ToastExample.show('Awesome', ToastExample.SHORT);` from JavaScript to display a short toast on the screen.
+We also recommend enabling [Gradle Daemon](https://docs.gradle.org/2.9/userguide/gradle_daemon.html) to speed up builds as you iterate on Java code.
 
-Create a new Java Class named `ToastModule.java` inside `android/app/src/main/java/com/your-app-name/` folder with the content below:
+### Create A Custom Native Module File
 
-```java
-// ToastModule.java
+The first step is to create the CalendarModule.java Java file inside `android/app/src/main/java/com/your-app-name/ folder`. This Java file will contain your native module Java class.
 
-package com.your-app-name;
+<figure>
+  <img src="/docs/assets/native-modules-android-add-class.png" width="700" alt="Image of adding a class called CalendarModule.java within the Android Studio." />
+  <figcaption>Image of how to add the CalendarModuleClass</figcaption>
+</figure>
 
-import android.widget.Toast;
+Then add the following content:
 
+```jsx
```

Comment:
Please change back the code block language from `jsx` to `java`.

Reply:
done